File: src/main/java/com/yupi/springbootinit/model/entity/MemberPriceConfig.java
package com.yupi.springbootinit.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import lombok.Data;

@TableName(value = "member_price_config")
@Data
public class MemberPriceConfig implements Serializable {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private String memberLevel;

    private String planType;

    private BigDecimal cashPrice;

    private Integer pointsPrice;

    private Integer durationDays;

    private String description;

    private String features;

    private Integer status;

    private Date createTime;

    private Date updateTime;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: web-admin/src/api/memberPriceConfig.ts
import request from './request';

export type MemberPlanType = 'month' | 'year' | 'lifetime';

export interface MemberPriceConfig {
  id: number;
  memberLevel: 'member';
  planType: MemberPlanType;
  cashPrice: number;
  currency: string;
  pointsPrice: 0;
  durationDays: number;
  description?: string;
  features?: string;
  status: number;
  createTime?: string;
  updateTime?: string;
}

export interface MemberPriceConfigUpdateRequest {
  id?: number;
  memberLevel?: 'member';
  planType?: MemberPlanType;
  cashPrice?: number;
  currency?: string;
  description?: string;
  features?: string;
  status?: number;
}

export async function listMemberPriceConfigs() {
  return request.get('/member-price-config/list') as Promise<{ data: MemberPriceConfig[] }>;
}

export async function updateMemberPriceConfig(params: MemberPriceConfigUpdateRequest) {
  return request.post('/member-price-config/update', params) as Promise<{ data: boolean }>;
}

export async function addMemberPriceConfig(params: MemberPriceConfigUpdateRequest) {
  return request.post('/member-price-config/add', params) as Promise<{ data: number }>;
}
